tests/Unit/QueryDataTableTest.php

```diff
@@ -28,9 +28,69 @@ public function test_complex_query_are_wrapped_and_countable()
         );
 
         $this->assertQueryWrapped(true, $dataTable->prepareCountQuery());
+        $this->assertQueryHasNoSelect(false, $dataTable->prepareCountQuery());
         $this->assertEquals(60, $dataTable->count());
     }
 
+    public function test_complex_query_use_select_in_count()
+    {
+        /** @var \Yajra\DataTables\QueryDataTable $dataTable */
+        $dataTable = app('datatables')->of(
+            DB::table('users')
+                ->select('users.*')
+                ->addSelect([
+                    'last_post_id' => DB::table('posts')
+                                        ->whereColumn('posts.user_id', 'users.id')
+                                        ->orderBy('created_at')
+                                        ->select('id'),
+                ])
+                ->orderBy(DB::table('posts')->whereColumn('posts.user_id', 'users.id')->orderBy('created_at')->select('created_at')
+                )
+        );
+
+        $this->assertQueryHasNoSelect(false, $dataTable->prepareCountQuery());
+        $this->assertEquals(20, $dataTable->count());
+    }
+
+    public function test_complex_query_can_ignore_select_in_count()
+    {
+        /** @var \Yajra\DataTables\QueryDataTable $dataTable */
+        $dataTable = app('datatables')->of(
+            DB::table('users')
+                ->select('users.*')
+                ->addSelect([
+                    'last_post_id' => DB::table('posts')
+                                        ->whereColumn('posts.user_id', 'users.id')
+                                        ->orderBy('created_at')
+                                        ->select('id'),
+                ])
+                ->orderBy(DB::table('posts')->whereColumn('posts.user_id', 'users.id')->orderBy('created_at')->select('created_at')
+                )
+        )->ignoreSelectsInCountQuery();
+
+        $this->assertQueryHasNoSelect(true, $dataTable->prepareCountQuery());
+        $this->assertEquals(20, $dataTable->count());
+    }
+
+    public function test_simple_queries_with_complexe_select_are_wrapped_without_selects()
+    {
+        /** @var \Yajra\DataTables\QueryDataTable $dataTable */
+        $dataTable = app('datatables')->of(
+            DB::table('users')
+              ->select('users.*')
+              ->addSelect([
+                  'last_post_id' => DB::table('posts')
+                                      ->whereColumn('posts.user_id', 'users.id')
+                                      ->orderBy('created_at')
+                                      ->select('id'),
+              ])
+        );
+
+        $this->assertQueryWrapped(true, $dataTable->prepareCountQuery());
+        $this->assertQueryHasNoSelect(true, $dataTable->prepareCountQuery());
```

Comment:
```suggestion
        $this->assertQueryHasNoSelect(false, $dataTable->prepareCountQuery());
```

Should use `ignoreSelectsInCountQuery` if want to have no select as true?

Reply:
If the query does not need to use the selects, it remove them, the `ignoreSelectsInCountQuery()` is use only to force this behavior (for example, when performing a subquery inside an order on the main query as in the test)